src/workers/cluster/managers/VersionStateManager.ts

```diff
@@ -0,0 +1,44 @@
+import { mapping } from '@cluster/utils';
+import { VarsTable } from '@core/types';
+import { UpdateType, Version, VersionManager } from '@hunteroi/versioning';
+
+const mapUpdateType = mapping.choice(
+    mapping.in<UpdateType>('major', 'minor'),
+    mapping.string.map<UpdateType>(() => 'patch')
+);
+
+export class VersionStateManager {
+    #version: Version;
+
+    public constructor(private readonly db: VarsTable) {
+        this.#version = new Version();
+    }
+
+    public async getVersion(): Promise<string> {
+        await this.refresh();
+        return this.#version.toString();
+    }
+
+    public async updateVersion(type: string): Promise<void> {
+        await this.refresh();
+        const manager = new VersionManager(this.#version);
+
+        const mapped = mapUpdateType(type);
+        if (!mapped.valid) {
+            throw new Error('Invalid update type');
+        }
+
+        manager.update(mapped.value);
+
+        await this.db.set('version', this.#version);
+    }
+
+    private async refresh(): Promise<void> {
```

Comment:
Dont really need this refresh and cache pattern I think. Move this to `getVersion` and make `updateVersion` call `getVersion`. Can get rid of the `#version` field then

Reply:
Then `getVersion` would return a `Version` instance instead of a string. But we don't need that outside the manager as we always display the version as a string. I could just change the `refresh` to something like `getFromStorage` and it would return the `Version` instance directly instead of assigning it to the private field.

What do you think?

Reply:
Yep youre right. `getFromStorage` sounds better!